source/npm/util.js

```diff
@@ -109,3 +111,57 @@ exports.checkIgnoreStrategy = ({files}) => {
 		`);
 	}
 };
+
+// New files added part of .npmignore or not part of files-attribute in "package.json"
+exports.checkNewFiles = async (newFiles, filesFromFileAttribute) => {
+	const rootDir = pkgDir.sync();
+	const npmignoreExists = fs.existsSync(path.resolve(rootDir, '.npmignore'));
+
+	let result = [];
+
+	if (filesFromFileAttribute) {
+		result = getFilesNotPartOfFileAttribute(filesFromFileAttribute, newFiles);
+	}
+
+	if (npmignoreExists) {
+		result = result.concat(await getFilesIgnoredByDotnpmignore(newFiles));
+	}
+
+	return result;
+};
+
+async function getFilesIgnoredByDotnpmignore(fileList) {
+	if (!Array.isArray(fileList)) {
+		throw new TypeError('expected array, but got {typeof fileList}');
+	}
+
+	const result = [];
+	const whiteList = await ignoreWalker({
+		path: pkgDir.sync(),
+		ignoreFiles: ['.npmignore']
+	});
+	for (const file of fileList) {
+		if (!whiteList.includes(file)) {
+			result.push(file);
+		}
+	}
+
+	return result;
+}
+
+function getFilesNotPartOfFileAttribute(filesFromFileAttribute, fileList) {
+	if (!Array.isArray(fileList)) {
+		throw new TypeError(`expected array, but got ${typeof fileList}`);
+	}
+
+	return fileList.filter(minimatch.filter(getGlobPattern(filesFromFileAttribute),
+		{matchBase: true}));
+}
+
+function getGlobPattern(filesFromFileAttribute) {
```

Comment:
I'd rename this to `getIgnoredFilesGlob`.

Reply:
implement your suggestion